projects/NeRF-Det/nerfdet/data_preprocessor.py

```diff
@@ -0,0 +1,576 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import math
+from numbers import Number
+from typing import Dict, List, Optional, Sequence, Tuple, Union
+
+import numpy as np
+import torch
+from mmdet.models import DetDataPreprocessor
+from mmdet.models.utils.misc import samplelist_boxtype2tensor
+from mmengine.model import stack_batch
+from mmengine.utils import is_seq_of
+from torch import Tensor
+from torch.nn import functional as F
+
+from mmdet3d.models.data_preprocessors.utils import multiview_img_stack_batch
+from mmdet3d.models.data_preprocessors.voxelize import (
+    VoxelizationByGridShape, dynamic_scatter_3d)
+from mmdet3d.registry import MODELS
+from mmdet3d.structures.det3d_data_sample import SampleList
+from mmdet3d.utils import OptConfigType
+
+
+@MODELS.register_module()
+class NeRFDetDataPreprocessor(DetDataPreprocessor):
+    """Points / Image pre-processor for point clouds / vision-only / multi-
```

Comment:
It should explain the difference between `NeRFDetDataPreprocessor` and the default `Det3DDataPreprocessor`.